src/include/flockmtl/model_manager/providers/handlers/azure.hpp

```diff
@@ -33,8 +31,8 @@ class AzureModelManager {
     }
 
     nlohmann::json CallEmbedding(const nlohmann::json &json, const std::string &contentType = "application/json") {
-        string url = "https://" + _resource_name + ".openai.azure.com/openai/deployments/" + _deployment_model_name +
-                     "/embeddings?api-version=" + _api_version;
+        std::string url = "https://" + _resource_name + ".openai.azure.com/openai/deployments/" +
```

Comment:
Should API Version be fixed as a const?

Reply:
I'm not sure about this is Sunny's implementation, and I still need to go through the providers' implementation and test them